suite-native/module-trading/src/selectors/exchangeSelectors.ts

```diff
@@ -98,3 +130,86 @@ export const selectGroupedExchangeQuotes = createMemoizedSelector(
     },
 );
+
+export const selectExchangeAccountsWithTokensSectionList = createExchangeMemoizedSelector(
+    [
+        selectVisibleDeviceAccounts,
+        selectTokenDefinitions,
+        selectCurrentFiatRates,
+        selectLocalCurrency,
+        state => state,
+    ],
+    (accounts: Account[], tokenDefinitions, fiatRates, localCurrency, state) =>
+        accounts
+            .map<SectionListData<MyAsset, Account>[number]>((account: Account) => {
+                const networkTokenDefinitions = getSimpleCoinDefinitionsByNetwork(
+                    tokenDefinitions,
+                    account.symbol,
+                );
+
+                const knownTokens = filterKnownTokens(
+                    networkTokenDefinitions,
+                    account.symbol,
+                    account.tokens ?? [],
+                );
+
+                const tokensWithBalance = knownTokens.filter(
+                    token => parseFloat(token?.balance ?? '0') > 0,
+                );
+
+                const tokens: MyAsset[] = tokensWithBalance.map(token => {
+                    const fiatRateKey = getFiatRateKey(
+                        account.symbol,
+                        localCurrency,
+                        token.contract as TokenAddress,
+                    );
+                    const rate = fiatRates?.[fiatRateKey]?.rate;
+                    const fiatBalance =
+                        rate && token.balance ? toFiatCurrency(token.balance, rate) : null;
+
+                    const tokenSymbol = selectAccountTokenSymbol(
+                        state as unknown as TokensRootState,
+                        account.key,
+                        token.contract as TokenAddress,
+                    );
+
+                    return {
+                        symbol: account.symbol,
+                        name: token.name ?? '',
+                        balance: token.balance ?? '0',
+                        fiatBalance,
+                        tokenSymbol,
+                        contract: token.contract as TokenAddress,
+                        cryptoId: toTokenCryptoId(account.symbol, token.contract),
+                    };
+                });
+
+                const accountAsset = {
+                    symbol: account.symbol,
+                    name: account.accountLabel || '',
+                    balance: account.formattedBalance,
+                    fiatBalance:
+                        getAccountFiatBalance({
+                            account,
+                            localCurrency,
+                            rates: fiatRates,
+                            shouldIncludeStaking: false,
+                            shouldIncludeTokens: false,
+                        }) ?? '0',
+                    cryptoId: getCoingeckoId(account.symbol) as CryptoId,
+                };
+
+                const assets: MyAsset[] = [
+                    ...(parseFloat(account.formattedBalance) > 0 ? [accountAsset] : []),
```

Comment:
nit: To work with the same data
```suggestion
                    ...(parseFloat(accountAsset.balance) > 0 ? [accountAsset] : []),
```

Reply:
Fixed in 2c5b5f9332